### ISSUE TITLE
Investigate incorrect permissions for certain tables in postgres

### ISSUE THREAD

**Issue by lbeaufort** [fecgov/openFEC#2927]
There appear to be some permissions errors with certain tables in postgres. We should investigate permissions for all postgres tables and materialized views, and the possibility of creating an unnumbered migration file to manage permissions going forward. An unnumbered migration file should run every time, allowing us to make changes more dynamically. 

**Example:** `public.ofec_sched_e`'s owner is `fec_api` while all the other `ofec` tables' owner is `fec`. As a result, only the `fec_api` role can access this table

**Completion criteria:**
- [ ] research an un-numbered migration file for managing permissions going forward in [data/migrations](https://github.com/18F/openFEC/tree/develop/data/migrations) - will Flyway run the migration each time we deploy if it’s unnumbered? Does it matter whether it's run first or last? 
- [ ] inventory permissions on existing tables and materialized views in postgres to make sure the `fec` role has read/write access and `fec_read` role has read access to them

**Comment by AmyKort** [comment]
@vrajmohan

**Comment by fecjjeng** [comment]
Currently, we have the following schema in database fec
('public','fecmur','aouser','auditsearch','disclosure','fecapp','staging','rad_pri_user','real_efile','real_pfile')

1. fec should be the owner of all tables, views, and MVs in these schemas and has full privilege of these objects.
2. fec_read should have SELECT and only SELECT privilege of all these fec owned tables, views, and MVs
3. aomur_usr has SELECT, INSERT, UPDATE, DELETE to all tables, views, and MVs in schemas fecmur and aouser
4, real_file has SELECT, INSERT, UPDATE, DELETE to all tables, views, and MVs in schemas real_efile and real_pfile
5. openfec_read is an old role, all members in this role had been granted fec_read role.  openfec_read will not be mainttained at this moment.  Considering drop it in the future.

**Comment by AmyKort** [comment]
Is this ready to close?

**Comment by fecjjeng** [comment]
Yes


On Mar 15, 2018, at 12:50 PM, Amy Kort <notifications@github.com<mailto:notifications@github.com>> wrote:


Is this ready to close?

**Comment by fecjjeng** [comment]
Some database objects have incorrect permissions, PR #3001 had a migration file had already address these permission issues. However, after the PR had been merged in DEV database and all permission are good, we noticed the permission to ofec_sched_e had been flipped back to the incorrect permission repeatedly while all the other fixed permission stays correct. After research, I discovered that the cronjob to refresh this table everyday is actually recreate this table. And the script that recreate this table does not specify the owner; and the cronjob is run by fec_api, so every night this table recreated, the ownership flip back to fec_api. Therefore, in addition to the migration file, data/refresh/rebuild_schedule_e.sql need to be updated to specify the ownership of this table.
